src/frontend/src/handler/drop_database.rs

```diff
@@ -55,6 +56,26 @@ pub async fn handle_drop_database(
         }
     };
 
+    // Check if database was created by an admin user, if so, require admin privilege to drop
+    let user_reader = session.env().user_info_reader().read_guard();
+    let current_user = user_reader
+        .get_user_by_name(&session.user_name())
+        .ok_or_else(|| {
+            ErrorCode::PermissionDenied("Session user is invalid".to_owned())
+        })?;
+
+    // If the database owner was an admin, only admin users can delete it
+    if let Some(database_owner_name) = user_reader.get_user_name_by_id(&database.owner()) {
+        if let Some(database_owner) = user_reader.get_user_by_name(&database_owner_name) {
+            if database_owner.is_admin && !current_user.is_admin && !current_user.is_super {
```

Comment:
Even if current user is super, it still can't drop a database that was created by another admin user.